p2p/net/swarm/dial_ranker.go

```diff
@@ -77,70 +75,92 @@ func DefaultDialRanker(addrs []ma.Multiaddr) []network.AddrDelay {
 	for i := 0; i < len(addrs); i++ {
 		res = append(res, network.AddrDelay{Addr: addrs[i], Delay: 0})
 	}
+
 	res = append(res, getAddrDelay(pvt, PrivateTCPDelay, PrivateQUICDelay, 0)...)
-	res = append(res, getAddrDelay(ip4, PublicTCPDelay, PublicQUICDelay, 0)...)
-	res = append(res, getAddrDelay(ip6, PublicTCPDelay, PublicQUICDelay, 0)...)
+	res = append(res, getAddrDelay(public, PublicTCPDelay, PublicQUICDelay, 0)...)
 	res = append(res, getAddrDelay(relay, PublicTCPDelay, PublicQUICDelay, relayOffset)...)
 	return res
 }
 
-// getAddrDelay ranks a group of addresses(private, ip4, ip6) according to the ranking logic
-// explained in defaultDialRanker.
+// getAddrDelay ranks a group of addresses according to the ranking logic explained in
+// documentation for defaultDialRanker.
 // offset is used to delay all addresses by a fixed duration. This is useful for delaying all relay
-// addresses relative to direct addresses
+// addresses relative to direct addresses.
 func getAddrDelay(addrs []ma.Multiaddr, tcpDelay time.Duration, quicDelay time.Duration,
 	offset time.Duration) []network.AddrDelay {
+
 	sort.Slice(addrs, func(i, j int) bool { return score(addrs[i]) < score(addrs[j]) })
 
+	// If the first address is (QUIC, IPv6), make the second address (QUIC, IPv4).
```

Comment:
Why does this only apply to QUIC? If a peer has QUIC disabled, we should still do Happy Eyeballs for it's TCP6 and TCP4 address, no?

Reply:
We can, provided we add a delay of 750ms between the two. I was not sure if we wanted that so I've done it just for QUIC.

Reply:
Ok, let's do this once https://github.com/libp2p/go-libp2p/issues/2394 is resolved then.